File: backend/src/main/java/ch/zhaw/students/gtd/entity/Task.java
package ch.zhaw.students.gtd.entity;

import lombok.Data;

import javax.persistence.*;
import java.util.Date;

@Entity
@Data
public class Task {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;

    private String description;

    @Temporal(TemporalType.TIMESTAMP)
    private Date dueDate;

    private int priority;

    private boolean done;

    private Long projectId;

    public Task() {
    }

    public Task(String name, String description, Date dueDate, int priority, boolean done, Long projectId) {
        this.name = name;
        this.description = description;
        this.dueDate = dueDate;
        this.priority = priority;
        this.done = done;
        this.projectId = projectId;
    }
}

File: frontend/src/model/task.ts
import {Project} from "@/model/project";

export enum Priority {
    HIGH = 3, MEDIUM = 2, LOW = 1
}

export interface Task {
    id?: number;
    projectId?: number;
    name?: string;
    done?: boolean;
    description?: string;
    dueDate?: Date;
    priority?: Priority;
}

